library/Icingadb/Web/Controller.php

```diff
@@ -349,19 +360,82 @@ public function createViewModeSwitcher()
             $viewModeSwitcher->setDefaultViewMode($viewMode);
         }
 
-        $viewModeSwitcher->populate(['view' => $this->params->shift($viewModeSwitcher->getViewModeParam())]);
+        $viewModeSwitcher->populate([
+            'view' => $this->params->shift($viewModeSwitcher->getViewModeParam())
+        ]);
 
-        $viewModeSwitcher->on(ViewModeSwitcher::ON_SUCCESS, function (ViewModeSwitcher $viewModeSwitcher) use ($prefs) {
-            $viewMode = $viewModeSwitcher->getValue($viewModeSwitcher->getViewModeParam());
+        if (
+            $viewModeSwitcher->getViewMode() === 'minimal'
+            || Url::fromRequest()->getParam(ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM) === 'minimal'
+        ) {
+            $hasLimitParam = Url::fromRequest()->hasParam($limitControl->getLimitParam());
 
-            $icingadbPrefs = $prefs->icingadb ?: [];
-            $icingadbPrefs['view_mode'] = $viewMode;
-            $prefs->icingadb = $icingadbPrefs;
+            if ($limitControl->getDefaultLimit() <= LimitControl::DEFAULT_LIMIT && ! $hasLimitParam) {
+                $limitControl->setDefaultLimit($limitControl->getDefaultLimit() * 2);
+            }
 
-            $this->redirectNow(Url::fromRequest()->setParam($viewModeSwitcher->getViewModeParam(), $viewMode));
-        });
+            if ($paginationControl->getDefaultPageSize() <= LimitControl::DEFAULT_LIMIT && ! $hasLimitParam) {
+                $paginationControl->setDefaultPageSize($paginationControl->getDefaultPageSize() * 2);
+
+                $paginationControl->apply();
+            }
+        }
+
+        $session = $this->Window()->getSessionNamespace(
+            self::SESSION_PREFIX . $this->Window()->getContainerId() . Url::fromRequest()->getPath()
+        );
 
-        $viewModeSwitcher->handleRequest(ServerRequest::fromGlobals());
+        // Clear session when the view mode param doesn't exists
+        if (! Url::fromRequest()->hasParam(ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM)) {
```

Comment:
We're in `createViewmodeSwitcher()` here and should be able to know exactly what the request's view mode is. And that's when we shift the view mode value to pass it to `populate()` above. Let's use the result here as well, shall we?